src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -53,38 +60,51 @@
 
     private final AuthorizationCheckService authCheckService;
 
+    private final FileUploadSubmissionRepository fileUploadSubmissionRepository;
+
     private final TokenProvider tokenProvider;
 
+    private final FileUploadExerciseRepository fileUploadExerciseRepository;
+
     public FileResource(FileService fileService, ResourceLoader resourceLoader, UserService userService, AuthorizationCheckService authCheckService,
-            LectureRepository lectureRepository, TokenProvider tokenProvider) {
+            LectureRepository lectureRepository, TokenProvider tokenProvider, FileUploadSubmissionRepository fileUploadSubmissionRepository,
+            FileUploadExerciseRepository fileUploadExerciseRepository) {
         this.fileService = fileService;
         this.resourceLoader = resourceLoader;
         this.userService = userService;
         this.authCheckService = authCheckService;
         this.lectureRepository = lectureRepository;
         this.tokenProvider = tokenProvider;
+        this.fileUploadSubmissionRepository = fileUploadSubmissionRepository;
+        this.fileUploadExerciseRepository = fileUploadExerciseRepository;
     }
 
     /**
      * POST /fileUpload : Upload a new file.
      *
      * @param file The file to save
      * @param keepFileName specifies if original file name should be kept
+     * @param isSubmission specifies if file is submitted by a student
      * @return The path of the file
      * @throws URISyntaxException if response path can't be converted into URI
      */
     @PostMapping("/fileUpload")
-    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR', 'TA')")
-    public ResponseEntity<String> saveFile(@RequestParam(value = "file") MultipartFile file, @RequestParam("keepFileName") Boolean keepFileName) throws URISyntaxException {
+    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR', 'TA', 'USER')")
+    public ResponseEntity<String> saveFile(@RequestParam(value = "file") MultipartFile file, @RequestParam("keepFileName") Boolean keepFileName,
+            @RequestParam(value = "isSubmission", required = false, defaultValue = "false") Boolean isSubmission) throws URISyntaxException {
         log.debug("REST request to upload file : {}", file.getOriginalFilename());
 
         // NOTE: Maximum file size is set in resources/config/application.yml
         // Currently set to 10 MB
 
+        if (isSubmission && file.getSize() > 2_000_000) {
```

Comment:
Please move this to a constant and make sure to write it as `2 * 1024 * 1024`.

Reply:
Moved it